### PR TITLE
#1021 - New-PnPTeamsTeam , ability to add multiple owners and members

### LINKED ISSUE #1021: [FEATURE] New-PnPTeam Allow Add Multiple Owners
**Is your feature request related to a problem? Please describe.**
Consider you have multiple users to add. If you pass them all to ```Owner``` parameter it would through an error below:
image
However the command would complete just fine if you pass a single user:
image

**Describe the solution you'd like**
Please make ```Owners``` parameter of ```New-TeamsTeam``` accept collections (aka arrays)

**Describe alternatives you've considered**

- ```Add-PnPMicrosoft365GroupOwner``` 

- ```Add-PnPTeamsUser```

### PR DESCRIPTION
**Pull request by gautamdsheth**
## Type ##
- [ ] Bug Fix
- [x] New Feature
- [ ] Sample

## Related Issues? ##
Fixes #1021  , #1022 

## What is in this Pull Request ? ##
Adds ability to add multiple owners and members to a Teams team.

### CODE REVIEW

**Comment by gautamdsheth** [comment]
Found a way to improve this PR. Will add some code tomorrow

File: src/Commands/Utilities/TeamsUtility.cs [hunk 1]
```diff
@@ -181,15 +191,23 @@ public static async Task<Team> NewTeamAsync(string accessToken, HttpClient httpC
...
-                var user = await GraphHelper.GetAsync<User>(httpClient, "v1.0/me?$select=Id", accessToken);
-                ownerId = user.Id;
+                if (owners != null && owners.Length > 0)
+                {
+                    var user = await GraphHelper.GetAsync<User>(httpClient, $"v1.0/users/{owners[0]}?$select=Id", accessToken);
+                    ownerId = user.Id;
+                }
+                else
+                {
+                    var user = await GraphHelper.GetAsync<User>(httpClient, "v1.0/me?$select=Id", accessToken);
```
[the remark this answers is not in the document]
Comment (gautamdsheth) [reply src/Commands/Utilities/TeamsUtility.cs]: Added a check to detect this connection type and based on that make a request. Hope that's ok.

File: src/Commands/Utilities/TeamsUtility.cs [hunk 2]
```diff
@@ -181,15 +191,23 @@ public static async Task<Team> NewTeamAsync(string accessToken, HttpClient httpC
...
             Group group = new Group();
             // get the owner if no owner was specified
             var ownerId = string.Empty;
             if (string.IsNullOrEmpty(owner))
             {
-                var user = await GraphHelper.GetAsync<User>(httpClient, "v1.0/me?$select=Id", accessToken);
-                ownerId = user.Id;
+                if (owners != null && owners.Length > 0)
+                {
+                    var user = await GraphHelper.GetAsync<User>(httpClient, $"v1.0/users/{owners[0]}?$select=Id", accessToken);
```
[the remark this answers is not in the document]
Comment (gautamdsheth) [reply src/Commands/Utilities/TeamsUtility.cs:228]: Added a fallback for this , will filter based on mail attribute.

**Comment by KoenZomers** [comment]
@gautamdsheth I have rebuilt the logic behind this one. Not sure how you got it to work the way it was. In my test cases it looks to work fine, but as this is likely an often used cmdlet, please check this code again as well. I've removed quite some pieces of what I believed was redundant code. Also removed the /me Graph call. Not sure if that could wreck backwards compatibility with this cmdlet.

**Comment by gautamdsheth** [comment]
@KoenZomers - tested it, fixed the chunk add logic because Graph only allows adding 20 users at a time. Much cleaner code now, thanks a lot !

**Comment by KoenZomers** [comment]
@gautamdsheth We're not there yet. With a fresh view I noticed that Owner and Owners are not mandatory. And I believe in the current situation that would mean that the current user executing the cmdlet would become the owner. In this updated could it throws an exception in that case:

image

Let me debug it (now I can finally debug again! 🥳) and see where it goes wrong.

**Comment by gautamdsheth** [comment]
@KoenZomers - can you give it a try now ? Have fixed the issue that you faced. If the case if app-only context, we will create a group without an owner.

**Comment by KoenZomers** [comment]
Pending PG validation if there's a maximum on the [Graph API](https://docs.microsoft.com/en-us/graph/api/conversationmembers-add?view=graph-rest-1.0&tabs=http) on how many users we can add in 1 call

**Pull request merged**